### PR TITLE
[web] Fix paragraph positioning

### PR DESCRIPTION
**Pull request by ferhatb**
When blendmode filters are applied to elements preceding paragraphs, the paragraph elements shift unless left,top css attributes are set to 0px.

Fixes: https://github.com/flutter/flutter/issues/56971
Added regression test.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/bitmap_canvas.dart [hunk 1]
```diff
@@ -714,6 +714,10 @@ class BitmapCanvas extends EngineCanvas {
       rootElement.append(paragraphElement);
     }
     _children.add(paragraphElement);
+    // If there is a prior sibling such as img prevent left/top shift.
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/bitmap_canvas.dart:717]: Is there any information about why this is happening? Would be useful to leave a link explaining what causes the shift.

Comment (ferhatb) [reply lib/web_ui/lib/src/engine/bitmap_canvas.dart:717]: No info or chromium bug i could point to. Although position is absolute and the paragraph has transform on it, it is shifting the text vertically down.

**Comment by ferhatb** [comment]
Merging on infra fail / mac android.

File: lib/web_ui/lib/src/engine/bitmap_canvas.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/bitmap_canvas.dart:717]: /me pulling my hair out 🤦

**Pull request merged**